### DISCUSSION TITLE
Logging on the tracing timeline

### DISCUSSION THREAD

**Discussion by GimpArm** [datalust/seq-tickets#2100] [Help]
I'm currently evaluating Seq for my company, we would like to use it as an OpenTelemetry exporter for our asp.net applications. I have it configured and working mostly how we will want it but I have one problem I cannot find a solution to. I know it is possible because the screen shots in your documentation show it.

When we create log messages the logs do not show up on the tracing timeline. They are part of the trace, they have the correct trace id, and show up in the events list with the correct time stamp but they are not displayed in the trace view.
Here is a screenshot of a trace with 2 log messages.
![Screenshot 2024-02-19 163444](https://github.com/datalust/seq-tickets/assets/11071591/52cdc4a8-5ad2-41ba-bbf2-73214f9c4069)

Could you give me a hint how to make this happen? Do I need to set a special attribute or add some kind of activity link?

**Comment by KodrAus** [comment]
Hi @GimpArm :wave: The current trace view is a visualization over the log data that (currently) just displays the span events themselves in the waterfall. It doesn't include the log messages.

We're currently planning our next release milestone though that'll include a round of enhancements over this first cut of the trace view so this is a helpful datapoint to have! If you run into anything else while experimenting with traces we'd love to get your feedback on it.

**Reply by SKProCH** [reply]
Hello, @KodrAus, is there any news about this feature?

**Reply by nblumhardt** [reply]
Hi @SKProCH! Yes, this feature was implemented in 2024.2 (I think?) and you can now see logs on the tracing timeline as small triangular markers:

![image](https://github.com/user-attachments/assets/0e1e3a56-8cb9-4f45-9338-d3d4eebc9df9)

**Reply by SKProCH** [reply]
Hello, @nblumhardt, yeah, I see this markers, and can hower over it and read the message, this is nice! But probably there is some way to "expand" this markers to full logs or something like that? Since associating it with actual log entries is time consuming currently

**Reply by liammclennan** [reply]
If you choose 'find' for the trace it will filter the event list to the spans and logs for that trace.

**Reply by nblumhardt** [reply]
:+1:

It's also possible to zoom in on them by clicking the span and choosing "Find".

Curious, though, are there any subtleties around how your spans and log events relate that we might be missing? We do use this quite a bit ourselves, but only tend to have a handful of events in each span - guessing that's not the case in your data?

**Reply by SKProCH** [reply]
No, probably about the same as everyone else.

I just think that perhaps integrating logs into trace view will allow analyzing data faster. For example, allowing you to "expand" these markers on the timeline to see the logs right there, like this:

![image](https://github.com/user-attachments/assets/0b0cb05d-1160-4166-a044-e15508b98801)

I assume that logs are probably not much different from spans from a user's perspective (they also contain text, and the "recording" date, parent SpanId), and I think it would be cool to be able to see them in the context of a trace.

Anyway, thank you very much for your answers and tips!

**Reply by nblumhardt** [reply]
Thanks for the follow-up, we appreciate all of the additional info.

**Comment by GimpArm** [comment]
Ok, thank you for the information. This would be an extremely helpful functionality to have for viewing traces.
